File: control-flow/Q105/main.go
package main

import "fmt"

func main() {
	var cost float64
	fmt.Print("Enter cost: ")
	fmt.Scan(&cost)
	for i := 0; i < 10; i++ {
		cost -= cost * 20 / 100
		fmt.Printf("cost for year %d: %f\n", i+1, cost)
	}
}

File: control-flow/Q105/main.rs
use std::io::{self, Read};

fn main() {
    let mut input = String::new();
    io::stdin().read_to_string(&mut input).expect("Failed to read input");

    let mut it = input.split_whitespace();
    let cost_str = it.next().unwrap_or("");
    let mut cost: f64 = cost_str.parse().expect("Not a valid number");

    for i in 0..10 {
        cost -= cost * 20.0 / 100.0;
        println!("cost for year {}: {}", i + 1, cost);
    }
}